### ISSUE TITLE
Question: Direct Messaging limits

### ISSUE THREAD

**Issue by grantjayy** [stanvanrooy/instauto#117]
Hey there,

Absolutely love this project! Well written and documented code, thanks for that!

I noticed the new release feature to Direct Message people on Instagram - this is an amazing game changer.

Instagram is very particular when it comes to messaging people... on IOS you're only allowed to message 90-150 people in a certain time frame, and once you hit that limit, after a 60-minute interval, 15 new message threads become available to you.

When it comes to this API Wrapper, what are the limitations with Messaging limits? Also, are you able to send a photo/video with the message? Are there any other warnings to take into account?

Thanks again for writing and sharing this!

**Comment by stanvanrooy** [comment]
Hey,

This project is a wrapper around the API used in the Android app, so the limit will be whatever the limit in the app is.

You can send all types of media, except for video. For that I need to finish #60.

See wiki: https://github.com/stanvanrooy/instauto/wiki/Using-direct-messaging

Thanks,
Stan

**Comment by grantjayy** [comment]
Awesome, thanks man!

**Comment by stanvanrooy** [comment]
No problem. Glad you like the package :)

Lmk if I can help with anything else.

**Comment by grantjayy** [comment]
I appreciate that. 

One thing I am confused on is how this package connects to the API through the Andriod app. After reading the code I'm not sure exactly how that's happening. Are you using the Graph API at all? Is it some sort of bug from Instagram?

I've been working with Instagram libraries for a few projects, particularly InstaPy, which is almost useless in my case now. This package however is perfect. So I am worried about the potential for this package to be closed/useless at some point because Instagram closes some loophole or exploit. 

Thanks again.

**Comment by stanvanrooy** [comment]
I believe InstaPy uses Selenium to emulate a browser session and use Instagram from the browser.

This package calls the Instagram API directly. It calls the same endpoints that the Instagram app calls.

We get these endpoints by sniffing the network traffic send from the Instagram API to the Instagram server.

Because of the backwards compatibility Instagram has to maintain with older versions of the app, they can't change the API endpoints whenever they want (something they can do in the web UI, for example).

That means that once a feature has been implemented correctly in `Instauto`, it will keep working without any new bugs for a good amount of time.

And if they do change an endpoint, which of course has happened before, you can just update the request Instauto sends.

tldr: this package calls the internal Instagram API, and that will always be possible, as long as an Instagram app exists :)

**Comment by grantjayy** [comment]
This is great. Really appreciate your help man!
